### PR TITLE
Overload Manager: LoadShedPoint for HCM decode headers

### PR DESCRIPTION
**Pull request by KBaichoo**
Commit Message: Implement a LoadShedPoint for HCM decode headers.
Additional Description:
Risk Level:  low / med
Testing: unit / integration
Docs Changes: included
Release Notes: included
Platform Specific Features: n/a
[Optional Runtime guard:] config (off by default)
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -343,6 +343,10 @@ new_features:
   change: |
     added load shed point ``envoy.load_shed_points.tcp_listener_accept`` that rejects new tcp connections
     by closing the connection before the new connection accept phase.
+- area: load shed point
+  change: |
+    added load shed point ``envoy.load_shed_points.http_connection_manager_decode_headers`` that rejects new http stream
```
Comment (RyanTheOptimist) [inline changelogs/current.yaml:348]: nit: "new http stream" => "new http streams", I think.

File: docs/root/configuration/operations/overload_manager/overload_manager.rst [hunk 2]
```diff
@@ -151,6 +151,11 @@ The following core load shed points are supported:
     - Envoy will reject (close) new TCP connections. This occurs before the
       :ref:`Listener Filter Chain <life_of_a_request>` is created.
 
+  * - envoy.load_shed_points.http_connection_manager_decode_headers
+    - Envoy will reject a new HTTP stream by sending a local reply. This occurs
```
Comment (RyanTheOptimist) [inline docs/root/configuration/operations/overload_manager/overload_manager.rst:155]: nit: "a new HTTP stream" -> "new HTTP streams"

**Review by RyanTheOptimist — COMMENTED** [review]
Two small nits

File: changelogs/current.yaml (hunk 1 above)
Comment (KBaichoo) [reply changelogs/current.yaml]: done

File: docs/root/configuration/operations/overload_manager/overload_manager.rst (hunk 2 above)
Comment (KBaichoo) [reply docs/root/configuration/operations/overload_manager/overload_manager.rst]: done

**Review by RyanTheOptimist — APPROVED** [review]
LGTM

**Review by adisuissa — COMMENTED** [review]
Thanks, small nit but otherwise LGTM.
/assign-from @envoyproxy/senior-maintainers

**Comment by KBaichoo** [comment]
/unassign @yanavlasov 

Ryan is already senior maintainer, just need cross company

File: test/integration/overload_integration_test.cc [hunk 3]
```diff
@@ -442,4 +447,40 @@ TEST_P(LoadShedPointIntegrationTest, ListenerAcceptShedsLoad) {
...
+
+  // Disable overload, Envoy should proxy the request.
+  updateResource(0.80);
+  test_server_->waitForGaugeEq(
+      "overload.envoy.load_shed_points.http_connection_manager_decode_headers.scale_percent", 0);
+
+  auto response_that_will_be_proxied =
+      codec_client_->makeHeaderOnlyRequest(default_request_headers_);
+  ASSERT_TRUE(response_that_will_be_proxied->waitForEndStream());
+  EXPECT_EQ(response_that_will_be_proxied->headers().getStatusValue(), "200");
```
[the remark this answers is not in the document]
Comment (KBaichoo) [reply test/integration/overload_integration_test.cc:483]: done

**Comment by KBaichoo** [comment]
/assign @zuercher 

for cross company review

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

**Comment by adisuissa** [comment]
Need to merge main.
@zuercher gentle ping, if you can look at it while Kevin is OOO.

**Comment by RyanTheOptimist** [comment]
@KBaichoo Looks like this needs a main merge again
@zuercher ping

**Review by zuercher — DISMISSED** [review]
Thanks!

**Review by yanavlasov — APPROVED** [review]

**Comment by yanavlasov** [comment]
Re-approving post merge

**Pull request merged**